### PR TITLE
Migrate Block Switcher Test case to Playwright

### PR DESCRIPTION
**Pull request by pavanpatil1**
## What?

Part of https://github.com/WordPress/gutenberg/issues/38851.
Migrate [block-switcher.test.js ](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/various/block-switcher.test.js) to its Playwright version.
## Why?

Part of https://github.com/WordPress/gutenberg/issues/38851.
## How?

See[ MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.
## Testing Instructions

Run` npm run test:e2e:playwright test/e2e/specs/editor/various/block-switcher.spec.js`

### CODE REVIEW

**Comment by kevin940726** [comment]
Hi @pavanpatil1! This is still on my backlog but I'm currently busy with something else. I'll get back to this as soon as possible! Thanks for the PR!

File: test/e2e/specs/editor/various/block-switcher.spec.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+	} ) => {
+		// Insert a list block.
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '- List content' );
+
+		await page.keyboard.press( 'ArrowUp' );
+		await pageUtils.pressKeys( 'alt+F10' );
+
+		// Verify the block switcher exists.
+		expect( page.locator( 'role=button[name="List"i]' ) ).toBeTruthy();
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/block-switcher.spec.js]: A locator is always truthy. Let's try `toBeVisible()` instead? We can also scope it under the block toolbar.

```suggestion
		const blockToolbar = page.getByRole( 'toolbar', { name: 'Block tools' } );
		await expect( blockToolbar.getByRole( 'button', { name: 'List' } ) ) ).toBeVisible();
```

File: test/e2e/specs/editor/various/block-switcher.spec.js [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+	test( 'Should show the expected block transforms on the list block when the quote block is removed', async ( {
+		page,
+		blockSwitcher,
+		pageUtils,
+	} ) => {
+		const wp = '';
+
+		// Remove the quote block from the list of registered blocks.
+		await page.evaluate( () => {
+			wp.blocks.unregisterBlockType( 'core/quote' );
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/block-switcher.spec.js]: We can do `window.wp` here so we don't need the wp const above.

File: test/e2e/specs/editor/various/block-switcher.spec.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+		// Insert a list block.
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '- List content' );
+
+		await page.keyboard.press( 'ArrowUp' );
+		await pageUtils.pressKeys( 'alt+F10' );
+
+		// Verify the block switcher exists.
+		expect( page.locator( 'role=button[name="List"i]' ) ).toBeTruthy();
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/block-switcher.spec.js]: Ditto.

File: test/e2e/specs/editor/various/block-switcher.spec.js [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+			] )
+		);
+	} );
+
+	test( 'Should not show the block switcher if all the blocks the list block transforms into are removed', async ( {
+		page,
+		pageUtils,
+	} ) => {
+		const wp = '';
+		async function getAvailableBlockTransforms() {
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/block-switcher.spec.js:89]: Why not use the function from the utils below?

File: test/e2e/specs/editor/various/block-switcher.spec.js [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+		// Insert a list block.
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '- List content' );
+
+		await page.keyboard.press( 'ArrowUp' );
+		await pageUtils.pressKeys( 'alt+F10' );
+
+		// Remove the paragraph and quote block from the list of registered blocks.
+		await page.evaluate( () => {
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/block-switcher.spec.js:104]: This is executed before the key press in the original test. I think we should follow the original behavior :).

File: test/e2e/specs/editor/various/block-switcher.spec.js [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+				'core/group',
+				'core/heading',
+				'core/columns',
+			].forEach( ( block ) => wp.blocks.unregisterBlockType( block ) );
+		} );
+
+		// Verify the block switcher exists.
+		expect(
+			page.locator(
+				'.block-editor-block-toolbar .block-editor-block-switcher'
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/block-switcher.spec.js]: Ditto.

File: test/e2e/specs/editor/various/block-switcher.spec.js [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+	async getAvailableBlockTransforms() {
+		await this.page.click( 'role=button[name="List"i]' );
+		return this.page.evaluate( ( buttonSelector ) => {
+			return Array.from(
+				document.querySelectorAll( buttonSelector )
+			).map( ( button ) => {
+				return button.textContent;
+			} );
+		}, '.block-editor-block-switcher__popover .block-editor-block-switcher__transforms__menugroup button' );
+	}
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/block-switcher.spec.js:198]: We can rewrite this into accessible selectors:

```js
await this.getBlockSwitcher( blockName ).click();
return await this.page.getByRole( 'menu', { name: blockName } )
	.getByRole( 'menuitem' )
	.allTextContents();
```

I left out the implementation of `getBlockSwitcher` here though. Feel free to ask questions if it's not clear!

**Review by kevin940726 — COMMENTED** [review]
Thank you! Sorry for the late review! This is close though! 💯

**Comment by Mamaduka** [comment]
Migrated via #57381.

**Pull request closed without merging**